File: Swift/task2.swift
/* 
Albert Astrom and Sam Polyakov
CS333
Project 3
task2.swift
*
TO RUN: swift task2.swift
*/
func binarySearch(_ arr: [Int],_ target: Int) -> Int? {
    var low = 0
    var high = arr.count - 1
    
    while low <= high {
        let mid = (low + high) / 2
        
        if arr[mid] == target {
            return mid
        } else if arr[mid] < target {
            low = mid + 1
        } else {
            high = mid - 1
        }
    }
    
    return nil
}

func find(arr: [Int], target: Int) {
    if let result = binarySearch(arr, target) {
    print("Found at index: \(result)")
    return
    } else {
    print("Not found.")
    return
}

}

let present = [1, 2, 3, 4, 5]
let notPresent = [100, 200, 300, 400, 500, 600]

find(arr: present, target: 3)
find(arr: notPresent, target: 3)

File: Rust/task2.rs
/* 
Albert Astrom and Sam Polyakov
CS333
Project 3
task2.rs
*
TO COMPILE: rustc task2.rs
TO RUN: ./task2
*/

fn bst(arr: &[i32], target: i32) ->  Option<usize> {
    // returns index of a value in a list using binary search
    let mut lowIndex = 0;
    let mut highIndex = arr.len()-1;

    while(lowIndex <= highIndex){
        let mut middleIndex = (lowIndex+highIndex)/2;
        if(arr[middleIndex] == target){
            return Some(middleIndex);
        }
        else if(arr[middleIndex] < target){
            lowIndex = middleIndex+1;
        }
        else{
            highIndex = middleIndex-1;
        }
    }
    None
}

fn main() {
    let arr = [1, 3, 5, 7, 9, 11, 13, 15, 17, 19];
    let target = 15;

    match bst(&arr, target) {
        Some(index) => println!("Found at index {}", index),
        None => println!("Not found"),
    }
}
